python/paddle/fluid/executor.py

```diff
@@ -1898,8 +1899,16 @@ def _run_using_fleet_executor(self,
                 rank_info.rank = rank
                 rank_info.ip_port = endpoint
                 fleet_exe_desc.cluster_info.append(rank_info)
+            nrank = len(trainer_endpoints)
         else:
             logging.warning("Fleet Executor will run on single device only.")
+        fleet_opt = program._pipline_opt["fleet_opt"]
```

Comment:
Where to set this attribute? User defined? Or we can obtain this somewhere automatically (like passing the DistributeStrategy object to the exe.run() method)?

Reply:
`_pipeline_opt` btw